runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -726,25 +743,39 @@ private void recoverRuntime() {
      * When this function returns, the maps are fully loaded.
      */
     public void loadMaps() {
-        log.info("loadMaps: Starting to resurrect maps");
-        initializeHeadAndTails();
-        recoverRuntime();
+        log.info("loadMaps: Starting to loadmaps");
 
-        log.info("loadMaps[startAddress: {}, stopAddress (included): {}, addressProcessed: {}]",
-                logHead, logTail, addressProcessed);
-        log.info("loadMaps: Loading successful, Corfu maps are alive!");
+        while (true) {
+            try {
+                recoverRuntime();
+                log.info("loadMaps [startAddress: {}, stopAddress (included): {}, addressProcessed: {}]",
+                        logHead, logTail, addressProcessed);
+                log.info("loadMaps: loading successful, Corfu maps are alive!");
+                return;
+            } catch (TrimmedException e) {
+                if (handleRetry() == false) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary comparisons in boolean expressions](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651460)